### PR TITLE
feat: disabling image lazy load for homepage articles

### LINKED ISSUE #969: Performance: avoid lazy-loading images above the fold
**Is your feature request related to a problem? Please describe.**
[Core Web Vitals](https://web.dev/vitals/) uses [Largest Contentful Paint](https://web.dev/lcp/) as an important metric for overall site performance. This metric measures the approximate time before the most important content on a page is visible to the user. There is evidence that lazy-loading images above the fold might negatively impact LCP scores, especially if the mobile viewport shows primarily a single image above the fold.

**Describe the solution you'd like**
This thread from the [AMP plugin](https://wordpress.org/support/topic/how-to-not-lazy-load-the-first-featured-image/) suggests some ways we can optimize above-the-fold images, namely: 

1. Disabling native WP lazy loading when AMP is on by setting the `loading` attribute to an empty string when calling `the_post_thumbnail` or `wp_get_attachment_image`, as AMP should already be doing its own server-side optimization
2. Adding a `data-hero-candidate` attribute to above-the-fold images
3. Incorporating [this filter](https://gist.github.com/westonruter/6adb64e6e8c858a40ac1dc51b03f16d8) to enable pre-loading of hero images

Ideally these optimizations should happen only on images rendered by blocks that appear toward the top of the post content, which might appear above the fold.

Use the [Web Vitals tool](https://web.dev/measure/) to measure performance impact before and after these optimizations.

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Implements Lazy Load toggle for the Homepage Posts block in order to improve [Largest Contentful Paint (LCP)](https://web.dev/i18n/en/lcp/) score for blocks above the fold.

It also implements by default the `data-hero-candidate` attribute to take advantage of AMP's prerendering features for LCP improvement.

Closes #969.

### How to test the changes in this Pull Request:

With AMP Plus on:

1. Check out this branch and install it on a JN or behind ngrok.
2. Set up a page with a Homepage Posts as the first block. It should use "Show media behind" as the media position to take as much screen space as possible. 
3. Visit the page and notice the `` contains the `loading="lazy"` attribute.
4. Obtain the page metrics at https://web.dev/measure
5. Edit the block on the "Featured Image Settings" toggle on the "Disable Featured Image Lazy Loading"
6. Visit the page and notice the `` contains the `loading="none"` attribute.
7. Obtain and compare the page metrics at https://web.dev/measure. Make sure to use an arbitrary `?v=2` parameter for the second report to skip any cache.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/blocks/homepage-articles/templates/article.php [hunk 1]
```diff
@@ -35,10 +35,17 @@ function( $data ) {
 		if ( has_post_thumbnail() && 'uncropped' !== $attributes['imageShape'] ) {
 			$image_size = Newspack_Blocks::image_size_for_orientation( $attributes['imageShape'] );
 		}
-		$thumbnail_args = '';
+		$thumbnail_args = array(
+			'data-hero-candidate' => true,
```
Comment (dkoo) [inline src/blocks/homepage-articles/templates/article.php:39]: Do we actually want to include this attribute for every image, or just the ones with `disableImageLazyLoad`?

**Review by dkoo — APPROVED** [review]
This looks good and works as described. On my test site it results in about 0.5s lower LCP and a ~5 point increase in the CWV Performance score!

Just have one question, which I'm not sure of myself. I don't really know the ramifications of including `data-hero-candidate` on every Homepage Posts thumbnail, but my gut says we should probably limit that attribute to only the images that are marked not to lazy-load?

File: src/blocks/homepage-articles/templates/article.php (hunk 1 above)
Comment (miguelpeixe) [reply src/blocks/homepage-articles/templates/article.php:39]: I wish I had a better understanding of the "candidate" portion of this attribute. I assume it makes the image eligible to become `data-hero` but controlled by a more complex logic.

As stated by @laurelfulford on https://github.com/Automattic/newspack-theme/pull/1616#issuecomment-991219146, it is standard for AMP 2.1+ to apply this attribute to any `the_post_thumbnail()`.

Comment (dkoo) [reply src/blocks/homepage-articles/templates/article.php:39]: > I wish I had a better understanding of the "candidate" portion of this attribute.

Me too! This is the only real explanation I could find: https://github.com/ampproject/amp-toolbox-php/pull/86

Sounds like it's okay to have multiple candidates—the later ones won't be used, but it shouldn't hurt performance.

**Pull request merged**